features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/RolePermissions/AdminTable/AdminTableBody.jsx

```diff
@@ -112,18 +112,17 @@ function AdminTableBody(props) {
         <TableBody>
             {stableSort(rows, getComparator(order, orderBy))
                 .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
-                .map((row) => {
-                    const key = row[0].key || row[0];
+                .map((row, rowIndex) => {
                     const isItemSelected = isSelected(row.name);
-                    const labelId = `enhanced-table-checkbox-${key}`;
+                    const labelId = `enhanced-table-checkbox-${rowIndex}`;
                     return (
                         <TableRow
                             hover
-                            onClick={(event) => handleClick(event, key)}
+                            onClick={(event) => handleClick(event, rowIndex)}
                             role='checkbox'
                             aria-checked={isItemSelected}
                             tabIndex={-1}
-                            key={key}
+                            key={rowIndex} // eslint-disable-line react/no-array-index-key
```

Comment:
Shall we try to find a better key other than the array index?

Reply:
Since there can be multiple scope mappings per role and the list gets updated when a new scope mapping is added we can use the index as a uuid.